FreeRTOS-Plus/Demo/coreMQTT_Windows_Simulator/MQTT_Serializer/DemoTasks/SerializerMQTTExample.c

```diff
@@ -0,0 +1,1208 @@
+ * of the packet and processes the packet based on the type.
+ *
+ * @param xMQTTSocket is a TCP socket that is connected to an MQTT broker to which
+ * an MQTT connection has been established.
+ */
+static void prvMQTTProcessIncomingPacket( Socket_t xMQTTSocket );
+
+/**
+ * @brief The transport receive wrapper function supplied to the MQTT library for
+ * receiving type and length of an incoming MQTT packet.
+ *
+ * @param[in] pxContext Pointer to network context.
+ * @param[out] pBuffer Buffer for receiving data.
+ * @param[in] bytesToRecv Size of pBuffer.
+ *
+ * @return Number of bytes received or zero to indicate transportTimeout;
+ * negative value on error.
+ */
+static int32_t prvTransportRecv( NetworkContext_t * pxContext,
+                                 void * pvBuffer,
+                                 size_t xBytesToRecv );
+
+/**
+ * @brief Generate and return monotonically increasing packet identifier.
+ *
+ * @return The next PacketId.
+ *
+ * @note This function is not thread safe.
+ */
+static uint16_t prvGetNextPacketIdentifier( void );
+
+/*-----------------------------------------------------------*/
+
+/* @brief Static buffer used to hold MQTT messages being sent and received. */
+static uint8_t ucSharedBuffer[ mqttexampleSHARED_BUFFER_SIZE ];
+
+/**
+ * @brief Packet Identifier generated when Subscribe request was sent to the broker;
+ * it is used to match received Subscribe ACK to the transmitted subscribe request.
+ */
+static uint16_t usSubscribePacketIdentifier;
+
+/**
+ * @brief Packet Identifier generated when Unsubscribe request was sent to the broker;
+ * it is used to match received Unsubscribe response to the transmitted unsubscribe
+ * request.
+ */
+static uint16_t usUnsubscribePacketIdentifier;
+
+/**
+ * @brief A pair containing a topic filter and its SUBACK status.
+ */
+typedef struct topicFilterContext
+{
+    const char * pcTopicFilter;
+    bool xSubAckSuccess;
+} topicFilterContext_t;
+
+/**
+ * @brief An array containing the context of a SUBACK; the SUBACK status
+ * of a filter is updated when a SUBACK packet is processed.
+ */
+static topicFilterContext_t xTopicFilterContext[ mqttexampleTOPIC_COUNT ] =
+{
+    { mqttexampleTOPIC, false }
+};
+
+
+/** @brief Static buffer used to hold MQTT messages being sent and received. */
+const static MQTTFixedBuffer_t xBuffer =
+{
+    .pBuffer = ucSharedBuffer,
+    .size    = mqttexampleSHARED_BUFFER_SIZE
+};
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief The Network Context implementation. This context is passed to the
+ * transport interface functions.
+ *
+ * This example uses transport interface function only to read the packet type
+ * and length of the incoming packet from network.
+ */
+struct NetworkContext
+{
+    Socket_t xTcpSocket;
+};
+/*-----------------------------------------------------------*/
+
+/*
+ * @brief Create the task that demonstrates the Serializer MQTT API Demo.
```

Comment:
"demonstrates the demo" seems like awkward phrasing
```suggestion
 * @brief Create the task that demonstrates the Serializer MQTT API.
```